src/Filecoin/Address.h

```diff
@@ -81,24 +113,28 @@ class Address {
     /// Returns ASCII character of type
     static char typeAscii(Type t) { return '0' + static_cast<char>(t); }
 
-    // Returns the payload size (excluding any prefixes) of an address type.
-    // If the payload size is undefined/variable (e.g. ID)
-    // or the type is unknown, it returns zero.
-    static uint8_t payloadSize(Type t) {
+    /// Validates if the payload size (excluding any prefixes and checksum) of an address type has an expected value.
+    static bool isValidPayloadSize(Type t, std::size_t payloadSize) {
         switch (t) {
-        case Type::SECP256K1:
-        case Type::ACTOR:
-            return 20;
-        case Type::BLS:
-            return 48;
-        default:
-            return 0;
+            case Type::ID:
+                return payloadSize == 0;
+            case Type::SECP256K1:
+            case Type::ACTOR:
+                return payloadSize == 20;
+            case Type::BLS:
+                return payloadSize == 48;
+            case Type::DELEGATED:
+                return payloadSize <= 54;
+            default:
+                return false;
         }
     }
 };
 
 inline bool operator==(const Address& lhs, const Address& rhs) {
-    return lhs.bytes == rhs.bytes;
+    return lhs.type == rhs.type
+           && lhs.actorID == rhs.actorID
+           && lhs.payload == rhs.payload;
 }
```

Comment:
is This function even used?

Reply:
I don't think so, it was, so I decided not to remove it

Reply:
Let's remove it if not used, will increase coverage

Reply:
Yes, it's used in tests
https://github.com/trustwallet/wallet-core/blob/8c0ff5c4cff8dff12ee7caa43707dcdb952b240f/tests/chains/Filecoin/AddressTests.cpp#L105-L109